### ISSUE TITLE
--reactor flag to trial does not work on Windows Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8745]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8745|
|Type|enhancement|
|Created|2016-08-11 12:57:00Z|
|Branch|https://github.com/twisted/twisted/tree/8745-rodrigc-reactors-2|

On Windows, with Python 3.5, I managed to
build the IOCP reactor, but I could not select it with trial.

```
python -m twisted.trial --reactor=iocp twisted
```

```
Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'win32er'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gi'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'iocp'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk2'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk3'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'wx'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'cf'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "C:\Users\Craig\twisted3\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'glib2'

C:\Users\Craig\twisted3\twisted\trial\__main__.py: The specified reactor does not exist: 'iocp'.
See the list of available reactors with --help-reactors
```

If I try to list the reactors, I get:

```
python -m twisted.trial --help-reactors
```

```
Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'win32er'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gi'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'iocp'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk2'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk3'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'wx'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'cf'

Unhandled Error
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\scripts\trial.py", line 601, in run
    config.parseOptions()
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 255, in parseOptions
    self._dispatch[optMangled](optMangled, arg)
  File "C:\Users\Craig\twisted3\twisted\python\usage.py", line 415, in <lambda>
    fn = lambda name, value=None, m=method: m()
  File "C:\Users\Craig\twisted3\twisted\application\app.py", line 500, in opt_help_reactors
    rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "c:\Users\Craig\twisted3\build\py35-tests-windows\lib\site-packages\zope\interface\interface.py", line 467, in _call_conform
    return conform(self)
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 71, in __conform__
    return self.load()
  File "C:\Users\Craig\twisted3\twisted\plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "C:\Users\Craig\twisted3\twisted\python\reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'glib2'

    default	A reasonable default: poll(2) if available, otherwise select(2).
    epoll	epoll(4)-based reactor.
    kqueue	kqueue(2)-based reactor.
    poll	poll(2)-based reactor.
    select	select(2)-based reactor.
```

<details><summary>Searchable metadata</summary>

```
trac-id__8745 8745
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8745_rodrigc_reactors_2 8745-rodrigc-reactors-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__None None
time__1470920220659761 1470920220659761
changetime__1484375883746029 1484375883746029
version__None None
owner__rodrigc rodrigc
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/663

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

How/why is Python 3 finding these reactor plugins?  They're in the `else` block if the `if _PY3` check of `src/twisted/plugins/twisted_reactors.py`?

I suspect this problem is triggered by weird .pyc / dropin.cache state and not something a regular user can encounter.

Can you give reproduction instructions?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for filing and submitting a fix for this, Craig.

I put some comments on https://github.com/twisted/twisted/pull/663.

I have reservations about the implementation strategy here, and I think the one tomprince voiced may be a critical problem.  I'm also curious to get an answer to exarkun's question - how is it that this bug is triggered?  I don't see a failing test (although I understand why it would be hard to write one for this).

Please add details, and possibly a different implementation, and resubmit for review.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc removed owner|
|-|-|

https://github.com/twisted/twisted/pull/675

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @rodrigc|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc commented|
|-|-|

In changeset 80ce0cc02e630a4a4e9d9e8b9e2266c4ed2ae3c6
```
#!CommitTicketReference repository="" revision="80ce0cc02e630a4a4e9d9e8b9e2266c4ed2ae3c6"
Merge pull request #675 from twisted/8745-rodrigc-reactors-2

Author: rodrigc
Reviewer: exarkun glyph
Fixes: ticket: ticket:1327 (#11332 after porting to GitHub) ticket:8745

Fix trial --help-reactors on Windows
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc set status to `closed`|
|-|-|
